app/src/main/java/com/duckduckgo/app/global/view/TypeAnimationTextView.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2019 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.global.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.widget.TextView
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+
+
+class TypeAnimationTextView @JvmOverloads constructor(
+    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
+) : TextView(context, attrs, defStyleAttr), CoroutineScope {
+
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Main + animationJob
+
+    private val animationJob: Job = Job()
+    private var typingAnimationJob: Job? = null
+
+    var typingDelayInMs: Long = 20
+    var delayAfterAnimationInMs: Long = 300
+    private lateinit var inputText: CharSequence
+
+    fun startTypingAnimation(inputText: CharSequence, isCancellable: Boolean = true, afterAnimation: () -> Unit = {}) {
+        this.inputText = inputText
+
+        if (isCancellable) {
+            setOnClickListener {
+                if ((typingAnimationJob as Job).isActive) {
```

Comment:
Ei @marcosholgado, I see that `typingAnimationJob` will be created right after this block of code, but what do you think about replacing the casting for this: `if (typingAnimationJob?.isActive == true)`

Just to avoid possible crashes if we modify this class and the order of how things are created. WDYT?

Reply:
Good shout! Done!